_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -2130,6 +2130,26 @@ struct FindNonSkip
 
     mfxU16 m_skipFrameMode;
 };
+mfxStatus FixForcedFrameType (DdiTask & newTask, mfxU32 frameOrder)
+{
+    if (frameOrder == 0 &&
+        !(newTask.m_type[0] & MFX_FRAMETYPE_IDR) &&
+        !(newTask.m_type[1] & MFX_FRAMETYPE_IDR))
+    {
+        if (newTask.m_type[0] & MFX_FRAMETYPE_I)
+            newTask.m_type[0] = newTask.m_type[0] | MFX_FRAMETYPE_IDR;
+        else if (newTask.m_type[1] & MFX_FRAMETYPE_I)
+            newTask.m_type[1] = newTask.m_type[1] | MFX_FRAMETYPE_IDR;
+        else
+            return MFX_ERR_UNDEFINED_BEHAVIOR;
+    }
+    if (!(newTask.m_type[0] & MFX_FRAMETYPE_REF))
+        newTask.m_type[0] = newTask.m_type[0] | MFX_FRAMETYPE_REF;
+    if (!(newTask.m_type[1] & MFX_FRAMETYPE_REF))
```

Comment:
No need to check condition

Reply:
formally condition is not needed, m_type[0] and m_type[1] should be reference for key frames. It is needed to avoid non necessary  operations = for fields of big structure. It helps while debug.

Reply:
Unconditional assignment is definitely better than condition checking followed either by assignment or by jump.
For debugging conditional breakpoints can be used instead.

Reply:
Your code is "if(a!=b) a=b;" I propose to change it to "a=b;".
It is cheaper, faster, simpler and less code.

Reply:
moreover, what will happen in progressive case? We will add MFX_FRAMETYPE_REF to type of non-existing field, as I see